scripts/globals/garrison.lua

```diff
@@ -485,16 +535,26 @@ local spawnNPC = function(zone, x, y, z, rot, name, look)
     return mob
 end
 
+-- Spawns all npcs for the zone in the given garrison starting npc
 local spawnNPCs = function(npc)
     local zone     = npc:getZone()
     local zoneData = xi.garrison.data[zone:getID()]
     local xPos     = zoneData.xPos
     local yPos     = zoneData.yPos
     local zPos     = zoneData.zPos
     local rot      = zoneData.rot
+
+    -- There's a bug on first time zone load where these allies array are null for some reason.
+    -- Since they are all the same for all zones, this is fine but we should really
+    -- find out why that's happening.
+    if zoneData.allies == nil then
+        print("[error] Allies array is nil. Will use default values.")
+        zoneData.allies = { "Trader", "Patrician", "Scholar" }
+    end
     local allyName = zoneData.allies[GetRegionOwner(zone:getRegionID()) + 1]
 
-    for i = 0, math.random(2, 6) do
+    -- Spawn 1 npc per player in alliance
+    for i = 1, #zoneData.players do
```

Comment:
I see 1 npc per ally player here: https://www.youtube.com/watch?v=GIIQdgCKdnY&t=80s.
All other captures with 1-3 players also always have the same npc amount per player (1 per player)